pubsub/gcp/pubsub/metadata.go

```diff
@@ -37,4 +37,7 @@ type metadata struct {
 	OrderingKey             string `mapstructure:"orderingKey"`
 	DeadLetterTopic         string `mapstructure:"deadLetterTopic"`
 	MaxDeliveryAttempts     int    `mapstructure:"maxDeliveryAttempts"`
+	MaxOutstandingMessages  int    `mapstructure:"maxOutstandingMessages"`
+	MaxOutstandingBytes     int    `mapstructure:"maxOutstandingBytes"`
+	NumGoroutines           int    `mapstructure:"numGoroutines"`
```

Comment:
I would prefer avoiding the use of terms like "goroutine" because they are more about the inner workings of Dapr (or Go in this case). What about "MaxParallelism" or "MaxConcurrentConnections" which are more closely aligned with the benefit for end users?

Reply:
Renamed to `MaxConcurrentConnections`